lib/v2/inewsweek/index.js

```diff
@@ -0,0 +1,38 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const iconv = require('iconv-lite');
+
+const rootUrl = 'http://news.inewsweek.cn';
+
+module.exports = async (ctx) => {
+    const channel = ctx.params.channel;
+    const url = `${rootUrl}/${channel}`;
+    const response = await got(url, {
+        responseType: 'buffer',
+    });
+    const $ = cheerio.load(iconv.decode(response.data, 'gbk'));
+    const items = $('div.grid-item')
+        .toArray()
+        .map(async (item) => {
+            item = $(item);
+            const href = item.find('a').attr('href');
+            const articleLink = `${rootUrl}${href}`;
+            const response = await got(articleLink, {
+                responseType: 'buffer',
+            });
+            const $$ = cheerio.load(iconv.decode(response.data, 'gbk'));
+            return {
+                title: item.find('p').text(),
+                description: $$('div.contenttxt').text(),
+                link: articleLink,
+                pubDate: parseDate(href.match('/.*/(.*)/.*')[1]),
+            };
```

Comment:
Please cache the returned result.

Reply:
Thanks a lot! The article items have been cached by using `tryGet`